synapse/lib/lmdbslab.py

```diff
@@ -154,8 +154,13 @@ def _growMapSize(self, size=None):
         return self.mapsize
 
     def initdb(self, name, dupsort=False):
+
         with self._noCoXact():
-            return self.lenv.open_db(name.encode('utf8'), dupsort=dupsort)
+            db = self.lenv.open_db(name.encode('utf8'), dupsort=dupsort)
+
+        self.dbdupsortmap[id(db)] = dupsort
```

Comment:
are we using id(db) here as the key because just db doesn't work for some reason?  could we just stuff it directly onto the db instance ?

Reply:
Changing to using a container object in order to do this.